google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/gcs/GcsBucketEditorProvider.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.gcs;
+
+import com.intellij.openapi.fileEditor.FileEditor;
+import com.intellij.openapi.fileEditor.FileEditorPolicy;
+import com.intellij.openapi.fileEditor.FileEditorProvider;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Provider of {@link GcsBucketContentEditor}'s for browsing Google Cloud Storage bucket contents.
+ */
+public final class GcsBucketEditorProvider implements FileEditorProvider {
+
+  @Override
+  public boolean accept(@NotNull Project project, @NotNull VirtualFile file) {
+    return file instanceof GcsBucketVirtualFile;
+  }
+
+  @NotNull
+  @Override
+  public FileEditor createEditor(@NotNull Project project, @NotNull VirtualFile file) {
+    return new GcsBucketContentEditor((GcsBucketVirtualFile) file);
+  }
+
+  @NotNull
+  @Override
+  public String getEditorTypeId() {
+    return "gcs-bucket-content-editor";
```

Comment:
Ditto about maybe needing a constant if this actually means something and needs to be unique or easily findable.

Reply:
from the docs:

> @return id of type of the editors that are created with this FileEditorProvider. Each FileEditorProvider should have unique non null id. The id is used for saving/loading of EditorStates.

I'll move it to a class-level constant

Reply:
done - i just made a private field. There isn't any sort of registry for these. Also the platform implementations of FileEditorProvider that I saw do the same thing.